### PR TITLE
Staffing logic improvements

### PR DESCRIPTION
**Pull request by marcua**
This PR does four things around staffing in StaffBot:
- A more efficient JOIN query in one of the staffing implementations
- Allows us to disable StaffBot inquiries for Workers who would otherwise be certified for a task using `staffbot_enabled`
- Allows us to priorities workers who receive StaffBot inquiries via `staffing_priority`
- Adds `settings.ORCHESTRA_ENFORCE_NO_NEW_TASKS_DURING_REVIEW` to toggle whether workers are eligible for more work if they currently have work returned from a reviewer.

### CODE REVIEW

File: example_project/example_project/orchestra_settings.py [hunk 1]
```diff
@@ -213,6 +213,10 @@ def setup_orchestra(settings_module_name):
     # Pass the Google Analytics key to templates with a context processor.
     install_context_processors(settings)
 
+    # Set to True if you want to block Workers from picking up new
+    # tasks while existing ones are returned by reviewers.
+    settings.ORCHESTRA_ENFORCE_NO_NEW_TASKS_DURING_REVIEW = True
```
Comment (joshblum) [inline example_project/example_project/orchestra_settings.py:218]: +1

**Comment by joshblum** [comment]
lgtm! So we'll manually toggle the staffbot enabled bool on workers? Maybe add a line to the docs about this and the priorities?

**Comment by marcua** [comment]
Yup that's right!  Just pushed docs

**Comment by joshblum** [comment]
lgtm yallah

**Pull request merged**